react-file-uploads/server/server.ts

```diff
@@ -0,0 +1,78 @@
+import 'dotenv/config';
+import express from 'express';
+import pg from 'pg';
+import {
+  ClientError,
+  errorMiddleware,
+  uploadsMiddleware,
+} from './lib/index.js';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.static('public'));
+app.use(express.json());
+
+app.post(
+  '/api/uploads',
+  uploadsMiddleware.single('image'),
+  async (req, res, next) => {
+    try {
+      if (!req.file) throw new ClientError(400, 'no file field in request');
+      const { caption } = req.body as Partial<Image>;
+      if (!caption) {
+        throw new ClientError(400, 'caption is a required field');
+      }
+      /* TODO:
+       * - create a url for the image by combining '/images/' with req.file.filename
+       * - insert the "caption" and "url" into the "images" table
+       * - respond with the inserted row data and the proper status code for inserting data
+       */
+      const url = `/images/${req.file.filename}`;
+      const sql = `
+      insert into "images" ("caption","url")
+      values($1, $2)
+      returning *;
+      `;
+      const params = [caption, url];
+      const result = await db.query<Image>(sql, params);
+      if (!result.rows[0]) {
```

Comment:
This is a good thing to think about, but the insert will always return a result (or throw an exception), so result.rows[0] will always exist.